### PR TITLE
Use std::regex to implement fnmatch()

### PR DESCRIPTION
**Pull request by joakim-hove**
Needed a break from manual writing

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/common/utility/shmatch.cpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    std::string re_pattern = pattern;
+
+
+    // Shell patterns should implicitly be interpreted as anchored at beginning
+    // and end.
+    re_pattern = "^" + re_pattern + "$";
+
+    // Shell wildcard '*' should be regular expression arbitrary character '.'
+    // repeated arbitrary number of times '*'
+    replace(re_pattern, "*", ".*");
```
Comment (blattms) [inline src/opm/common/utility/shmatch.cpp:51]: Any particular reason not to use e,g.
```
std::regex re("\\*");
re_pattern = std::regex_replace(repattern, re, ".*");
```

Comment (joakim-hove) [reply src/opm/common/utility/shmatch.cpp:51]: That looks fine - I'll update

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/common/utility/shmatch.cpp [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <fnmatch.h>
```
Comment (magnesj) [inline src/opm/common/utility/shmatch.cpp:20]: Why is include of `fnmatch.h` required? Easier to compile on Windows if not present.

Comment (joakim-hove) [reply src/opm/common/utility/shmatch.cpp:20]: Ahhh mistake - removing that was the whole point. Fox tomorrow!

**Pull request merged**